v0.9.0 Edge Support (#15)
maticnetwork/terraform-polygon-supernets: README.md

   AWS console. To learn more about setting up AWS credentials, check
   out the documentation provided by AWS
   [here](https://docs.aws.amazon.com/cli/latest/userguide/cli-chap-getting-started.html). `aws
   configure`, `aws configure sso`, or set appropriate variables in
   `~/.aws/credentials` or in `~/.aws/config`. You can directly set
   access keys like below.

```
$ export AWS_ACCESS_KEY_ID=AKIAIOSFODNN7EXAMPLE
$ export AWS_SECRET_ACCESS_KEY=wJalrXUtnFEMI/K7MDENG/bPxRfiCYEXAMPLEKEY
$ export AWS_DEFAULT_REGION=us-west-2
```

Note: The role of the AWS user should have permissions to create all
of the resources provided in `/modules`.

It's worthwhile at this point to confirm that AWS CLI is authenticated
properly. Try running this command `aws sts get-caller-identity`. If
this works, you're probably set to move forward.

3. There are a few environment variables that should be set before
   provisioning the infrastructure. Edit `example.env` based on your
   requirements and then run the commands below to set your local
   environment variables.

```
set -a
source example.env
set +a
```

You can quickly double check that your environment variables have been

> set by running `env | grep -i tf_var` to double check that the

Reviewer: Aims to be more direct (so the user does check) and removes "double check" redundancy"

environment variables match what you specified in `example.env`

The full list of variables that can be configured are in
[variables.tf](./variables.tf).

4. Now we'll run `terraform init` to download and install the provider
   plugins, which are used to interact with AWS and initializes the
   backend to store the state file.

```
terraform init
```

5. Now we're going to plan and apply using Terraform. While working
   iteratively on your supernet, you'll typically make changes to the
   Terraform modules, then plan to see what changes will be made to
   your infrastructure, and then finally apply if the plan looks good.

```
terraform plan
terraform apply
```

6. During the Terraform apply, a private key was generated to give you
   access to your VMs. You'll need to shave that key using the
   following commands.

```
terraform output pk_ansible > ~/devnet_private.key
chmod 600 ~/devnet_private.key
```

> At this point all of the AWS infrastructure needed to run a supernet

Reviewer: ```suggestion
By now, of the necessary AWS infrastructure for operating a Supernet should be deployed.
```

> should be deployed. This is a good time to login to your AWS Console

Reviewer: ```suggestion
It's a good time to sign in to your AWS Console and examine the setup.
```

> and take a look.

Reviewer: ```suggestion
```

## Ansible Deployment Steps

> 1. At this stage we'll be using Ansible to configure the VMs that we

Reviewer: The first statement here could probably go before all the steps are outlined as a general intro to the section.

>    just deployed with Terraform. First, Change working directory to

Reviewer: ```suggestion
   just deployed with Terraform. To begin, switch your working directory to the Ansible folder
```

   `ansible`.

```
cd ansible
```

> 2. The Ansible playbooks that we use have some external

Reviewer: ```suggestion
2. The Ansible playbooks that we use require certain external
```

>    dependencies. In order to retrieve those collections, you can run

Reviewer: ```suggestion
   dependencies. In order to retrieve these collections, you can run
```

   the following command:

```
ansible-galaxy install -r requirements.yml
```

> 3. If you've changed your region, company name, or deployment name,

Reviewer: ```suggestion
3. In case you've altered your region, company name, or deployment name, remember to
```

>    you'll need to modify `inventory/aws_ec2.yml`

Reviewer: ```suggestion
   modify the `inventory/aws_ec2.yml` file accordingly.
```

```
regions:
  - us-west-2
###
filters:
  tag:BaseDN: "<YOUR_DEPLOYMENT_NAME>.edge.<YOUR_COMPANY>.private"
```

> 4. The file `local-extra-vars.yml` has a bunch of values that are

Reviewer: ```suggestion
4. The `local-extra-vars.yml` file contains a number of values that are
```

>    commonly tweaked during supernet deployments. At the very least,

Reviewer: ```suggestion
   often adjusted during Supernet deployments. At the very least,
```

>    you'll probably need to change `clean_deploy_title` to match

Reviewer: ```suggestion
   make sure to update the `clean_deploy_title` to match
```

>    whatever deployment name you used.

Reviewer: ```suggestion
   the deployment name you used.
```

>    Some of the other values like `block_gas_limit` and `block_time`

Reviewer: ```suggestion
   Other values like `block_gas_limit` and `block_time`
```

>    are important for performance and are worth setting very carefully.

Reviewer: ```suggestion
   significantly impact performance and should be set with care.
```

> 5. Check if your instances are available by running the following.

Reviewer: ```suggestion
5. To verify the availability of your instances, run the following command:
```

```
ansible-inventory --graph
```

> This command will list a bunch of instances. It's important at this

Reviewer: ```suggestion
This command will list a bunch of instances. At this stage, it's important
```

stage to make sure that your SSM and SSH are configured to be able to

> reach these instances based on their instance id. To test SSM, run a

Reviewer: ```suggestion
to these instances based on their instance IDs. To test SSM,
```

> command like this:

Reviewer: ```suggestion
execute a command similar to the one below:
```

```
aws ssm start-session --target [aws instance id]
```

An example instance id is `i-0641e8ff5f2a31647`. When you run this

> command you should have shell access to your VM.

Reviewer: ```suggestion
command, you should have shell access to your VM.
```

> 6. Now that we know we can access our VMs, we need to ensure the

Reviewer: ```suggestion
6. Now that we've confirmed access to our VMs, we must verify that the instances are accessible by Ansible.
```

   instances are reachable by Ansible. Try running this command:

```
ansible --inventory inventory/aws_ec2.yml --extra-vars "@local-extra-vars.yml" all -m ping
```

7. Run ansible playbook
```
ansible-playbook --inventory inventory/aws_ec2.yml --extra-vars "@local-extra-vars.yml" site.yml
```

> After this full playbook runs you should have a functional

Reviewer: ```suggestion
After the **full playbook runs, you should have a functional
```

> supernet. If you want to do some tests, you can get your public rpc

Reviewer: ```suggestion
Supernet**. If you want perform some tests, you can obtain your public RPC
```
